src/monospace/index.ts

```diff
@@ -0,0 +1,126 @@
+import fetch from "node-fetch";
+
+import { FirebaseError } from "../error";
+import { loadRC } from "../rc";
+
+import type { GetInitFirebaseResponse, InitFirebaseResponse } from "./interfaces";
+
+/**
+ * Integrate Firebase Plugin with Monospace’s service Account Authentication
+ */
+export async function setupMonospace(projectRoot: string, project?: string): Promise<void> {
+  const initFirebaseResponse = await initFirebase(project);
+
+  if (initFirebaseResponse.success === false) {
+    throw new Error(String(initFirebaseResponse.error));
+  }
+
+  const { rid } = initFirebaseResponse;
+
+  // Poll for response from the user
+  const authorizedProject = await pollAuthorizedProject(rid);
+
+  createFirebaseRc(projectRoot, authorizedProject);
```

Comment:
Do we have a separate path where I can skip this? It looks like the only function I have control over calling is requireAuth() so it seems like you'd have to add a param to requireAuth() and pass it all the way down here.

Reply:
> Do we have a separate path where I can skip this? It looks like the only function I have control over calling is requireAuth() so it seems like you'd have to add a param to requireAuth() and pass it all the way down here.

currently we don't. Can you somehow set an environment variable so we can detect that it's running in the plugin?

Reply:
Added a condition to return the authorized project if it's coming from the extension(through `options.isVSCE`)